### PR TITLE
Allow creating new artifacts from artifact swap UI

### PR DESCRIPTION
**Pull request by arquelion**
## Describe your changes
In `ArtifactSwapModal.tsx`:
- Added "Add New Artifact" button to the artifact swap UI underneath the page selector which opens up the artifact editor with the slot selector disabled - This maintains parity with the rest of the site where the general order of buttons is filters then page/sort option selectors and then the "Add New" button.

In `ArtifactEditor.tsx`:
- Added `fixedSlotKey` as an optional prop for `ArtifactEditor` - This is used to set the slot if the slot dropdown is disabled.
- Modified the logic surrounding assigning `slotKey` and `newValue.slotKey` (inside `update`) to account for the slot dropdown being disabled

## Issue or discord link

- Resolves https://github.com/frzyc/genshin-optimizer/issues/1252

## Testing/validation
Current UI with the "Add New Artifact" button
image
Artifact Editor when opened through the artifact swap UI
image
image
image
image
image
Database showing the artifacts that were added
image

## Checklist before requesting a review (leave this PR as draft if any part of this list is not done.)

- [x] I have commented my code in hard-to understand areas.
- [x] I have made corresponding changes to README or wiki.
- [x] For front-end changes, I have updated the corresponding English translations.
- [x] I have run `yarn run mini-ci` locally to validate format and lint.
- [x] If I have added a new library or app, I have updated the deployment scripts to ignore changes as needed

### CODE REVIEW

**Comment by arquelion** [comment]
Converting to draft, I hadn't realized disabling the slot in the artifact editor currently locks artifact creation to flowers only, which isn't quite the behavior I want.

(Alternatively, I don't bother locking the slot at all and someone can just create a full set in one go and then go through the individual slots to equip the new artifacts.)

**Comment by arquelion** [comment]
Fixed, should be ready for review.

The Randomize button currently bypasses slot locking in the artifact editor, not going to bother with it since it's a dev-mode only feature anyway.

**Comment by frzyc** [comment]
https://genshin-optimizer-prs.github.io/pr/1501/frontend/

File: apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/ArtifactSwapModal.tsx [hunk 1]
```diff
@@ -187,6 +205,14 @@ export default function ArtifactSwapModal({
               showingTextProps={showingTextProps}
             />
           </Box>
+          <Button
+            fullWidth
+            onClick={onShowEditor}
+            color="info"
+            startIcon={<Add />}
+          >
+            {t('addNew', { ns: 'artifact' })}
```
Comment (frzyc) [inline apps/frontend/src/app/PageCharacter/CharacterDisplay/Tabs/TabOverview/ArtifactSwapModal.tsx:214]:
```suggestion
            {t('artifact:addNew')}
```

**Review by frzyc — APPROVED** [review]
LGTM, thanks for testing other usecases as well. The artifact editor logic is kind of bloated, and as we add more use cases from different places, the logic gets more complicated. 

I think we can look into removing `disableSlot` in favor of `fixedSlotKey` prop in a future PR.

**Comment by arquelion** [comment]
Committed your syntax change, this PR should be good to go then.

> I think we can look into removing disableSlot in favor of fixedSlotKey prop in a future PR.

I haven't lined up more issues from the list to tackle yet, so I can definitely look at this next.

**Review by frzyc — APPROVED** [review]

**Pull request merged**